app/models/pageflow/revision.rb

```diff
@@ -91,6 +93,14 @@ def find_file(model, id)
       UsedFile.new(file)
     end
 
+    def share_providers
+      self[:share_providers] || entry.theming.default_share_providers
+    end
+
+    def active_share_providers
+      share_providers.select { |_k, v| v }.keys
+    end
+
```

Comment:
Since `active_share_providers` is now a public API that is used in plugins, it should have a test.